controllers/storagecluster/external_resources_test.go

```diff
@@ -557,3 +557,52 @@ func TestExternalMonitoringResources(t *testing.T) {
 		})
 	}
 }
+
+func TestErasureCodedExternalResources(t *testing.T) {
+	cr := &api.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ocsinit",
+			Namespace: "openshift-storage",
+		},
+		Spec: api.StorageClusterSpec{
+			ExternalStorage: api.ExternalStorageClusterSpec{
+				Enable: true,
+			},
+			Monitoring: &api.MonitoringSpec{
+				ReconcileStrategy: string(ReconcileStrategyIgnore),
+			},
+		},
+	}
+	externalResource := []ExternalResource{
+		{
+			Name: "ceph-rbd",
+			Kind: "StorageClass",
+			Data: map[string]string{
+				"pool": generateNameForCephBlockPool(cr),
+			},
+		},
+		{
+			Name: "ceph-rbd-ec",
+			Kind: "StorageClass",
+			Data: map[string]string{
+				"dataPool": "ec-data-pool",
+				"pool":     "replicated-metadata-pool",
+			},
+		},
+	}
+
+	for _, extR := range externalResource {
+		t.Run(extR.Name, func(t *testing.T) {
+			actualSC := newCephBlockPoolStorageClassConfiguration(cr)
+			// To override the values for external cluster
+			for k, v := range extR.Data {
+				actualSC.storageClass.Parameters[k] = v
+			}
+			assert.NotEmpty(t, actualSC.storageClass.Parameters["clusterID"])
+			assert.Equal(t, extR.Data["dataPool"], actualSC.storageClass.Parameters["dataPool"])
```

Comment:
Where do you test that the `dataPool` name is `ec-data-pool`?

Reply:
`"dataPool": "ec-data-pool",` this what I'm passing as JSON blob

Reply:
Yes and it should be in the SC parameters, so the unit test should validate this.

Reply:
maybe I'm missing something but I'm passing `"dataPool": "ec-data-pool"` in the JSON blob and checking it with storage class parameters `actualSC.storageClass.Parameters["dataPool"])` if provided is equal to expected.